test/info.spec.ts

```diff
@@ -0,0 +1,183 @@
+  ExpressionInfoCollector,
+  FunctionCall,
+  RuntimeValue,
+  standardMemberChecks,
+} from '../src';
+
+describe('Info collector', () => {
+  it('analyses simple expressions without context', () => {
+    const info = new ExpressionInfoCollector({});
+
+    let res = info.collect('a');
+
+    expect(res.functionCalls).toHaveLength(0);
+    expect(res.errors).toHaveLength(0);
+
+    res = info.collect('b()');
+
+    expect(res.functionCalls).toHaveLength(1);
+    expect(res.errors).toHaveLength(0);
+
+    expect(res.functionCalls[0].name).toBe('b');
+    expect(res.functionCalls[0].args).toHaveLength(0);
+    expect(res.functionCalls[0].path).toHaveLength(0);
+
+    res = info.collect('a.b');
+
+    expect(res.functionCalls).toHaveLength(0);
+    expect(res.errors).toHaveLength(0);
+
+    res = info.collect('a.b.c()');
+
+    expect(res.functionCalls).toHaveLength(1);
+    expect(res.errors).toHaveLength(0);
+
+    expect(res.functionCalls[0].name).toBe('c');
+    expect(res.functionCalls[0].args).toHaveLength(0);
+    expect(res.functionCalls[0].path).toStrictEqual(['a', 'b']);
+
+    res = info.collect('a + c');
+
+    expect(res.functionCalls).toHaveLength(0);
+    expect(res.errors).toHaveLength(0);
+
+    res = info.collect('a * (b && c(1, 2))');
+
+    expect(res.functionCalls).toHaveLength(1);
+    expect(res.errors).toHaveLength(0);
+
+    expect(res.functionCalls[0].name).toBe('c');
+    expect(res.functionCalls[0].args).toStrictEqual([1, 2]);
+    expect(res.functionCalls[0].path).toHaveLength(0);
+  });
+
+  it('analyses simple expressions with simple context', () => {
+    const info = new ExpressionInfoCollector({
+      evalOpts: {
+        context: {
+          a: '__a__',
+          b: '__b__',
+        },
+      },
+    });
+
+    let res = info.collect('a');
+
+    expect(res.functionCalls).toHaveLength(0);
+    expect(res.errors).toHaveLength(0);
+
+    res = info.collect('b()');
+
+    expect(res.functionCalls).toHaveLength(1);
+    expect(res.errors).toHaveLength(0);
+
+    expect(res.functionCalls[0].name).toBe('b');
+    expect(res.functionCalls[0].args).toHaveLength(0);
+    expect(res.functionCalls[0].path).toHaveLength(0);
+
+    res = info.collect('a.b');
+
+    expect(res.functionCalls).toHaveLength(0);
+    expect(res.errors).toHaveLength(1);
+
+    res = info.collect('a.b.c()');
+
+    expect(res.functionCalls).toHaveLength(1);
+    expect(res.errors).toHaveLength(1);
+
+    expect(res.functionCalls[0].name).toBe('c');
+    expect(res.functionCalls[0].args).toHaveLength(0);
+    expect(res.functionCalls[0].path).toStrictEqual(['a', 'b']);
+
+    res = info.collect('a + c');
+
+    expect(res.functionCalls).toHaveLength(0);
+    expect(res.errors).toHaveLength(1);
+
+    res = info.collect('a * (b && c(1, 2))');
+
+    expect(res.functionCalls).toHaveLength(1);
+    expect(res.errors).toHaveLength(1);
+
+    expect(res.functionCalls[0].name).toBe('c');
+    expect(res.functionCalls[0].args).toStrictEqual([1, 2]);
```

Comment:
Should probably check 'a' and 'b' are valid too

Reply:
They aren't functions in the expression, so nothing is collected for them. They also exist in the context, so there are no errors for them either.